src/signer.rs

```diff
@@ -100,42 +104,57 @@ struct CngSigner {
 }
 
 impl CngSigner {
+    // new hash function using BCryptHash function which uses FIPS certified SymCrypt
     fn hash(&self, message: &[u8]) -> Result<(Vec<u8>, SignaturePadding), Error> {
-        let (hash, padding) = match self.scheme {
-            SignatureScheme::RSA_PKCS1_SHA256 => (
-                sha2::Sha256::digest(message).to_vec(),
-                SignaturePadding::Pkcs1,
-            ),
-            SignatureScheme::RSA_PKCS1_SHA384 => (
-                sha2::Sha384::digest(message).to_vec(),
-                SignaturePadding::Pkcs1,
-            ),
-            SignatureScheme::RSA_PKCS1_SHA512 => (
-                sha2::Sha512::digest(message).to_vec(),
-                SignaturePadding::Pkcs1,
-            ),
-            SignatureScheme::RSA_PSS_SHA256 => (
-                sha2::Sha256::digest(message).to_vec(),
-                SignaturePadding::Pss,
-            ),
-            SignatureScheme::RSA_PSS_SHA384 => (
-                sha2::Sha384::digest(message).to_vec(),
-                SignaturePadding::Pss,
-            ),
-            SignatureScheme::RSA_PSS_SHA512 => (
-                sha2::Sha512::digest(message).to_vec(),
-                SignaturePadding::Pss,
-            ),
-            SignatureScheme::ECDSA_NISTP256_SHA256 => (
-                sha2::Sha256::digest(message).to_vec(),
-                SignaturePadding::None,
-            ),
-            SignatureScheme::ECDSA_NISTP384_SHA384 => (
-                sha2::Sha384::digest(message).to_vec(),
-                SignaturePadding::None,
-            ),
+        let (alg, padding) = match self.scheme {
+            SignatureScheme::RSA_PKCS1_SHA256 => {
+                (BCRYPT_SHA256_ALG_HANDLE, SignaturePadding::Pkcs1)
+            }
+            SignatureScheme::RSA_PKCS1_SHA384 => {
+                (BCRYPT_SHA384_ALG_HANDLE, SignaturePadding::Pkcs1)
+            }
+            SignatureScheme::RSA_PKCS1_SHA512 => {
+                (BCRYPT_SHA512_ALG_HANDLE, SignaturePadding::Pkcs1)
+            }
+            SignatureScheme::RSA_PSS_SHA256 => (BCRYPT_SHA256_ALG_HANDLE, SignaturePadding::Pss),
+            SignatureScheme::RSA_PSS_SHA384 => (BCRYPT_SHA384_ALG_HANDLE, SignaturePadding::Pss),
+            SignatureScheme::RSA_PSS_SHA512 => (BCRYPT_SHA512_ALG_HANDLE, SignaturePadding::Pss),
```

Comment:
Consistent formatting?

Reply:
it is done by cargo fmt. I will change the original format in previous file.